### PR TITLE
add remove() unit tests

### PR DESCRIPTION
**Pull request by paddykontschak**
## Unit tests

- `remove()` on existent element
- `remove()` on non-existent element

## Expected result (test 2)
`remove(".missing")` should check if the requested element exists and if it doesn't, return a message saying so

## Current result (test 2)
`remove(".missing")` doesn't check if the requested element exists and returns an error:
```
TypeError: Cannot read property 'parentNode' of null
```
mojave results

## About the tests
There are 2 unit tests for the non-existent element. This is because `remove()` doesn’t check if an element exists and then fails shortly after trying to get that element’s parentNode.
The second unit test has that check added manually.

## Additional
I’m not entirely sure how to check if this function did what it did. So I just used `find()` to see if we can find it. I feel like we should add another check before the removal function to see if it existed in the first place.

As of right now, that’s not really an issue; as previously mentioned `remove()` fails when the element doesn’t exist. Once that is fixed and it stops failing it would probably make sense to add an additional check. Or ask for a return directly in the function.

Signed-off-by: Patrick Kontschak <pk@becklyn.com>

### CODE REVIEW

File: tests/cases/dom/manipulate/remove.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        } else {
+            removed = false;
+        }
+
+        const result = find(".missing");
+
+        assert.ok(false === removed, "couldn't remove element, element doesn't exist");
+        assert.equal(result.length, 0, ".missing not found");
+    }
+);
```
Comment (apfelbox) [inline tests/cases/dom/manipulate/remove.js]: The unit tests are for finding bugs and issues in the code. They should highlight flaws in the original code.

The goal is to have green tests with correct and complete test data. If one of these correct / complete tests fails, no workaround to the test should be added, but the failing test should be left as is.

Comment (apfelbox) [inline tests/cases/dom/manipulate/remove.js]: so... please remove this test.

File: tests/cases/dom/manipulate/remove.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    {
+        const find = mojave.dom.traverse.find;
+        const element = document.querySelector(".missing");
+        const removing = mojave.dom.manipulate.remove;
+
+        removing(element);
+
+        const result = find(".missing");
+
+        assert.equal(result.length, 0, ".missing not found");
```
Comment (apfelbox) [inline tests/cases/dom/manipulate/remove.js]: Please use:

```js
const before = found(".missing");
remove(before);
const after = found(".missing");

assert.equal(before.length, 0, "before: nothing found");
assert.equal(after.length, 0, "after: still nothing found");
```

Comment (apfelbox) [inline tests/cases/dom/manipulate/remove.js]: Same in the "existent element" test

File: tests/cases/dom/manipulate/remove.js (hunk 1 above)
Comment (apfelbox) [inline tests/cases/dom/manipulate/remove.js]: Also add an explicit test with `remove(null)` and `remove([])` here, that it doesn't throw.

**Comment by apfelbox** [comment]
I'll fix the last issues up in a follow up commit. Thanks!
